_source/_posts/2023-08-30-passwordless-authentication.md

```diff
@@ -0,0 +1,203 @@
+
+The web app for the site displays a form to gather user information, an email address, or a user name. The user enters the required information and presses **Register**. The web app recognizes the browser as FIDO2 capable and asks if the user wants to sign in using a passkey instead of a password. Of course, the user says, "Yes!"
+
+{% img blog/passwordless-authentication/webauthnio-register.jpg alt:"webauthn.io site showing a user name entered in the form and a modal from the browser asking to create a passkey" width:"800" %}{: .center-image }
+
+The web app calls the server with credential request options, such as the username and registration-related data. You can open debugging capabilities in your browser and watch the network requests. Here, we see the payload for the call, which includes the username.
+
+{% img blog/passwordless-authentication/webauthnio-register-payload.jpg alt:"webauthn.io payload to register a user includes the username along with registration metadata configurable in advanced settings, such as encryption algorithm" width:"800" %}{: .center-image }
+
+The server responds with information about the relying party, a user ID, and a cryptographically random value for the registration process called a "challenge." 
+
+{% img blog/passwordless-authentication/webauthnio-register-response.jpg alt:"webauthn.io response includes relying party id, user name and id, and the challenge" width:"800" %}{: .center-image }
+
+The user selects a device for the passkey, like a phone or tablet. The web app uses WebAuthn to generate a new credential, sending the response information from the server, including the challenge. The credential call from the browser to the authenticator is possible via CTAP2.
+
+The authenticator takes over, prompting the user for consent and biometric information. The authenticator generates a public/private key set for the website. The authenticator signs the public key with an attestation certificate for the device hardware. Signing the key with the attestation certificate proves the keyset is from a specific device. The authenticator returns the public key in the WebAuthn credential response.
+
+{% img blog/passwordless-authentication/webauthnio-ctap.jpg alt:"webauthn.io site with a modal to create a passkey showing a QR code" width:"800" %}{: .center-image }
+
+The web app calls the server with the public key and the user ID to store the user's credentials.
+
+{% img blog/passwordless-authentication/webauthnio-register-success-payload.jpg alt:"webauthn.io site network call payload to complete registration" width:"800" %}{: .center-image }
+
+The server verifies the key for completeness and integrity, such as ensuring the certificate's validity, the website origin is accurate, and that the challenge value matches the original response. Once verified, the server persists this info, matching your username with the public key so it has the information needed to authenticate you for the passkey later.
```

Comment:
"ensuring certificate's validity" is probably the right level of abstraction for here but it makes me curious which actual checks are involved -- check expiry? look up in a CA? look up in system CA, or a specific one?